playwright/src/test/java/com/microsoft/playwright/TestScreencast.java

```diff
@@ -76,15 +76,13 @@ void saveAsShouldThrowWhenNoVideoFrames(@TempDir Path videosDir) {
       page.close();
 
       Path saveAsPath = videosDir.resolve("my-video.webm");
-      try {
+      // WebKit pauses renderer before win.close() and actually writes something.
+      if (isWebKit()) {
         popup.video().saveAs(saveAsPath);
-      } catch (PlaywrightException e) {
-        // WebKit pauses renderer before win.close() and actually writes something.
-        if (isWebKit()) {
-          assertTrue(Files.exists(saveAsPath));
-        } else {
-          assertTrue(e.getMessage().contains("Page did not produce any video frames"), e.getMessage());
-        }
+        assertTrue(Files.exists(saveAsPath));
+      } else {
+        PlaywrightException e = assertThrows(PlaywrightException.class, () -> popup.video().saveAs(saveAsPath));
```

Comment:
The test [failed](https://github.com/microsoft/playwright-java/runs/8121694827?check_suite_focus=true) on one of the bots:
![image](https://user-images.githubusercontent.com/9798949/187961046-24927819-ce39-4984-8910-e122b8d2c48a.png)

Looks like there is a race between popup closure and us calling saveAs on the video. We may need to wait for the popup to finish closing?

```java
if (!popup.isClosed()) {
    popup.waitForClose(() -> {});
}
```

Reply:
done